transport_nantes/mailing_list/views.py

```diff
@@ -252,9 +254,57 @@ class MailingListListView(ListView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        lists = self.get_queryset()
-        context['mailing_lists'] = [(list, user_subscribe_count(list))
-                                    for list in lists if not list.is_petition]
-        context['petitions_lists'] = [(list, subscriber_count(list))
-                                      for list in lists if list.is_petition]
+        mailing_lists = self.get_queryset()
+        context['mailing_lists'] = [(mailing_list, user_subscribe_count(
+            mailing_list)) for mailing_list in mailing_lists
+            if not mailing_list.is_petition]
+        return context
+
+
+class UserStatusView(LoginRequiredMixin, ListView):
+    model = MailingList
+    template_name = "mailing_list/mailing_list_user_status.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['mailing_lists'] = list()
+        context['petitions_lists'] = list()
+        base_lists = MailingList.objects.filter(
+            list_active=True).order_by(
+                'is_petition', 'mailing_list_name')
+        for mailing_list in base_lists:
+            state = user_current_state(
+                self.request.user, mailing_list).event_type
+            if not mailing_list.is_petition:
+                context['mailing_lists'].append((mailing_list, state,))
+            else:
+                if state != "sub":
+                    context['petitions_lists'].append((mailing_list, state,))
+        return context
+
+
+class MailingListToggleSubscription(LoginRequiredMixin, FormView):
```

Comment:
Doing a GET on this URL (ml/toggle_subscription) gets you to a page that looks like this : 
![image](https://user-images.githubusercontent.com/70256364/156374717-79652af0-d365-4655-90db-00866eef04bd.png)

Maybe you want to redirect to /ml/status if no mailing list is given. :slightly_smiling_face: